python/ray/data/datasource/file_based_datasource.py

```diff
@@ -217,6 +220,11 @@ def expand_paths(
     ) -> Tuple[List[str], List[Optional[int]]]:
         from pyarrow.fs import FileType
 
+        logger.warning(
```

Comment:
```suggestion
       if len(paths) > 1:
           logger.warning(
```

This should hide the warning for the common case, where a single directory is passed as the path.

Reply:
Another option that I was considering here is to only log this warning after a fixed amount of time has elapsed expanding directory paths (e.g. if we've been expanding directories for more than 15 seconds, then log this warning).

Reply:
Ok, it seems fine to keep this simple since this is a precaution.